homeassistant/components/madvr/binary_sensor.py

```diff
@@ -0,0 +1,110 @@
+"""Binary sensor entities for the madVR integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import MadVRConfigEntry
+from .const import DOMAIN
+from .coordinator import MadVRCoordinator
+from .entity import MadVREntity
+
+_HDR_FLAG = "hdr_flag"
+_OUTGOING_HDR_FLAG = "outgoing_hdr_flag"
+_POWER_STATE = "power_state"
+_SIGNAL_STATE = "signal_state"
+
+
+@dataclass(frozen=True)
+class MadvrBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describe madVR binary sensor entity."""
+
+    value_fn: Callable[[MadVRCoordinator], bool] = lambda _: False
+
+
+BINARY_SENSORS: tuple[MadvrBinarySensorEntityDescription, ...] = (
+    MadvrBinarySensorEntityDescription(
+        key=_POWER_STATE,
+        translation_key=_POWER_STATE,
+        value_fn=lambda coordinator: coordinator.client.is_on
+        if coordinator.client
+        else False,
+    ),
+    MadvrBinarySensorEntityDescription(
+        key=_SIGNAL_STATE,
+        translation_key=_SIGNAL_STATE,
+        value_fn=lambda coordinator: bool(
+            coordinator.data.get("is_signal", False) if coordinator.data else False
+        ),
+    ),
+    MadvrBinarySensorEntityDescription(
+        key=_HDR_FLAG,
+        translation_key=_HDR_FLAG,
+        value_fn=lambda coordinator: bool(
+            coordinator.data.get("hdr_flag", False) if coordinator.data else False
+        ),
+    ),
+    MadvrBinarySensorEntityDescription(
+        key=_OUTGOING_HDR_FLAG,
+        translation_key=_OUTGOING_HDR_FLAG,
+        value_fn=lambda coordinator: bool(
+            coordinator.data.get("outgoing_hdr_flag", False)
+            if coordinator.data
+            else False
+        ),
```

Comment:
I think we can assume coordinator.data and coordinator.client are always there

Reply:
will fail if its removed

Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 598, in _async_add_entities
    await coro
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 912, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/workspaces/core/homeassistant/helpers/entity.py", line 1366, in add_to_platform_finish
    self.async_write_ha_state()
  File "/workspaces/core/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
    self._async_write_ha_state()
  File "/workspaces/core/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
    self.__async_calculate_state()
  File "/workspaces/core/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/entity.py", line 1011, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/binary_sensor/__init__.py", line 293, in state
    if (is_on := self.is_on) is None:
                 ^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/madvr/binary_sensor.py", line 97, in is_on
    return self.entity_description.value_fn(self.coordinator)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/madvr/binary_sensor.py", line 60, in <lambda>
    coordinator.data.get("outgoing_hdr_flag", False)
    ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'

Reply:
When is coordinator.data None?

Reply:
pushed fix